cpu/stm32f1/periph/spi.c

```diff
@@ -210,27 +211,37 @@ int spi_transfer_byte(spi_t dev, char out, char *in)
             return -1;
     }
 
-    while (!(spi->SR & SPI_SR_TXE)) {}
-    spi->DR = out;
-    transferred++;
-
-    while (!(spi->SR & SPI_SR_RXNE)) {}
-    if (in != NULL) {
-        *in = spi->DR;
-        transferred++;
+    if(!in){
+        for (unsigned i = 0; i < length; i++) {
+            while (!(spi->SR & SPI_SR_TXE)) {}
+            spi->DR = (uint8_t)out[i];
+            transferred++;
+        }
+        /* SPI busy */
+        while ((spi->SR & SPI_SR_BSY)) {}
+        spi->DR;
+    }
+    else if(!out) {
+        for (unsigned i = 0; i < length; i++) {
+            spi->DR = 0;
+            while (!(spi->SR & SPI_SR_RXNE)) {}
+            in[i] = (char)spi->DR;
+            transferred++;
+        }
     }
     else {
-        spi->DR;
+        for (unsigned i = 0; i < length; i++) {
+            while (!(spi->SR & SPI_SR_TXE)) {}
+            spi->DR = out[i];
+            transferred++;
+            while (!(spi->SR & SPI_SR_RXNE)) {}
+            in[i] = (char)spi->DR;
+        }
     }
 
-    /* SPI busy */
-    while ((spi->SR & 0x80)) {}
 #if ENABLE_DEBUG
     if (in != NULL) {
-        DEBUG("\nout: %x in: %x transferred: %x\n", out, *in, transferred);
-    }
-    else {
-        DEBUG("\nout: %x in: was nullPointer transferred: %x\n", out, transferred);
+        DEBUG("\nSPI: transferred %i Bytes\n", transferred);
```

Comment:
One last comment (sorry @PeterKietzmann), we could actually get rid of `transferred` since at this point `transferred` will always be `length`. Its super minor but would save 4 bytes :-)